### PR TITLE
connector/saml: log info at debug level

### PR DESCRIPTION
**Pull request by estroz**
connector/saml: log as much connection info for debugging as possible (without exposing sensitive information)

Logging connection data during query flow will give SAML users a debugging trace.

### CODE REVIEW

File: connector/saml/saml.go [hunk 1]
```diff
@@ -143,6 +143,7 @@ func (c *Config) openConnector(logger logrus.FieldLogger) (*provider, error) {
 		if f.val == "" {
 			missing = append(missing, f.name)
 		}
+		logger.Debugf("saml: field %q = %q", f.name, f.val)
```
Comment (ericchiang) [inline connector/saml/saml.go]: We might just want to include this as the initial logs we spit out in `cmd`

```
$ ./bin/dex serve examples/config.yaml 
time="2017-08-04T17:08:39Z" level=info msg="config issuer: https://docprod-radhika.dev.coreos.systems/identity" 
time="2017-08-04T17:08:39Z" level=info msg="config storage: memory" 
time="2017-08-04T17:08:39Z" level=info msg="config static client: tectonic-console" 
time="2017-08-04T17:08:39Z" level=info msg="config static client: tectonic-kubectl" 
time="2017-08-04T17:08:39Z" level=info msg="config connector: okta" 
time="2017-08-04T17:08:39Z" level=info msg="config connector: local passwords enabled" 
time="2017-08-04T17:08:39Z" level=info msg="config skipping approval screen" 
```

File: connector/saml/saml.go [hunk 2]
```diff
@@ -252,16 +258,21 @@ func (p *provider) POSTData(s connector.Scopes, id string) (action, value string
...
 		r.Issuer = &issuer{Issuer: p.entityIssuer}
+		logStr = fmt.Sprintf("saml: authn request entity issuer set to %q", p.entityIssuer)
 	}
+	p.logger.Debugln(logStr)
 
 	data, err := xml.MarshalIndent(r, "", "  ")
 	if err != nil {
 		return "", "", fmt.Errorf("marshal authn request: %v", err)
 	}
+	p.logger.Debugln("saml: marshalled authn request successfully")
```
Comment (ericchiang) [inline connector/saml/saml.go]: probably a little too verbose :)

File: connector/saml/saml.go [hunk 3]
```diff
@@ -252,16 +258,21 @@ func (p *provider) POSTData(s connector.Scopes, id string) (action, value string
 		},
 		AssertionConsumerServiceURL: p.redirectURI,
 	}
+
+	logStr := "saml: no authn request entity issuer set"
```
Comment (ericchiang) [inline connector/saml/saml.go]: Don't care for the `logStr` pattern. Probably fine to just have two log lines and an if-else.

**Comment by estroz** [comment]
Closing this in favor of a more comprehensive approach.

**Pull request closed without merging**